src/domainobjects/cash_balance.py

```diff
@@ -1,32 +1,61 @@
-from domainobjects.generatable  import Generatable
+from domainobjects.generatable import Generatable
 from datetime import datetime
 import random
 
 class CashBalance(Generatable):
-    
-    def generate(self, record_count, custom_args):
-        config = self.get_object_config()
-        records_per_file = config['max_objects_per_file']
-        file_num = 1
-
-        records = []        
-
-        for i in range(1, record_count+1):
-            records.append({
-                'amount': self.generate_random_integer(),
-                'curr': self.generate_currency(),
-                'account_num': self.generate_random_integer(length=8),
-                'purpose': self.generate_purpose(),
-                'time_stamp': datetime.now(),
-            })
-
-            if (i % int(records_per_file) == 0):
-                self.write_to_file(file_num, records)
-                file_num += 1
-                records = []
-
-        if records != []:
-            self.write_to_file(file_num, records)
-    
+    """ Class to generate cash balances. Generate method will generate a set
+    amount of balances. Other generation methods included where cash balances
+    are the only domain object requiring them. """
+
+    CASH_BALANCE_PURPOSES = ['Cash Balance', 'P&L', 'Fees']
+
+    def generate(self, record_count, start_id):
+        """ Generate a set number of cash balances
+
+        Parameters
+        ----------
+        record_count : int
+            Number of cash balances to generate
+        start_id : int
+            Starting id to generate from
+
+        Returns
+        -------
+        List
+            Containing 'record_count' cash balances
+        """
+
+        records = []
+
+        for _ in range(start_id, start_id+record_count):
+            records.append(self.generate_record())
+
+        return records
+
+    def generate_record(self):
+        """ Generate a single cash balance
+
+        Returns
+        -------
+        dict
+            A single cash balance object
+        """
+
+        return {
+            'amount': self.generate_random_integer(),
+            'currency': self.generate_currency(),
+            'account_num': self.generate_random_integer(length=8),
```

Comment:
Should use generate_account. 

Also accounts should be domain objects so if there isn't an issue for this alreday lets make it. Account Type will be part of the domain object so we probably don't need to have 'ICP' or 'ECP' as part of the accountId